### DISCUSSION TITLE
Proposal to create a utility to determine the appropriate Stockfish compilation for a specific processor

### DISCUSSION THREAD

**Discussion by Coolchessguykevin** [official-stockfish/Stockfish#4749] [Ideas]
Dear Community,

not all users know which Stockfish compilations are the best for their processor (#4748). And even advanced users are not always sure what choice to make it the best (x86-64, x86-64-vnni512, x86-64-vnni256, x86-64-sse41-popcnt, x86-64-bmi2, x86-64-avx512, x86-64-avx2?).
 
Of course, there is an option to test each compilation on your own, but this will not always be objective and the difference in speed may not be noticeable, taking into account the measurement error.
 
I propose to make a special simple utility, when launched, the user will receive an informational message with information like "For your computer and CPU, the most optimal Stockfish compilation is the following (BMI, for example)".
 
I think this utility will be very, very useful for many users.
 
Regards, Ivan.

**Comment by cj5716** [comment]
Hi, there is a makefish.sh script here https://github.com/official-stockfish/Stockfish/wiki/Compiling-from-source that automates this process of choosing arch.

**Reply by Coolchessguykevin** [reply]
Hello, 

I don't understand programming and I don't even know how to use it. I suggest creating some utility (executable file), with a simple launch of which any user will be able to find out which compilation is best to download from Github https://github.com/official-stockfish/Stockfish/releases page for his processor.

**Reply by Disservin** [reply]
You can download the windows stockfish downloader from here https://github.com/ppigazzini/stockfish-downloader#script-windows . And then double clicking it should be enough. It downloads the best one for your cpu directly

**Reply by Coolchessguykevin** [reply]
Thanks, didn't know there was such a way! I downloaded and ran this script and nothing happened. Nothing at all. The script cmd window instantly closed after launch and that's it. The downloaded Stockfish did not appear in the folder with the script. Running with administrator rights did not have any other effect. Checked on two computers.
 
However, I manually copied and pasted the cmd code in the Windows PowerShell and it worked! This is what I was needed, thanks again! Now I see the optimal versions for my processors (all archives downloaded at my PC User folder)!

It would be really cool if this method was added to the official release of Stockfish.

**Reply by vondele** [reply]
FYI @ppigazzini
